tinyagentos/worker/agent.py

```
@@ -793,6 +879,21 @@ async def run(self):
+                        # pre-restart checkpoint, run the health-check and
+                        # signal the outcome (success or rollback) to the
+                        # controller.
+                        #
+                        # Scheduled as a background task so the worker loop
+                        # can register and send heartbeats immediately; the
+                        # hook's grace period + health check would otherwise
+                        # block the loop long enough for the controller to
+                        # mark the worker offline (CodeRabbit, Jul 31).
+                        asyncio.ensure_future(_run_post_update_hook(
+                            self.controller_url, self, self._state_dir,
+                        ))
```

Ruff (0.16.1)

[warning] Store a reference to the return value of `asyncio.ensure_future`

(RUF006)